src/test/java/org/pesmypetcare/webservice/controller/appmanager/MyPetCareRestControllerTest.java

```diff
@@ -0,0 +1,127 @@
+package org.pesmypetcare.webservice.controller.appmanager;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule;
+import com.google.firebase.auth.FirebaseAuthException;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.pesmypetcare.webservice.error.ErrorBody;
+import org.pesmypetcare.webservice.securingservice.FirebaseFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.http.MediaType;
+import org.springframework.test.web.servlet.MockMvc;
+
+import static org.springframework.test.util.AssertionErrors.assertEquals;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@SpringBootTest
+@AutoConfigureMockMvc
+class MyPetCareRestControllerTest {
+    private static String jsonUser1;
+    private static String jsonUser2;
+    private static ObjectMapper objectMapper;
+    private static String url;
+    private static String key;
+    private static String password;
+    private static boolean deleteUser1;
+
+    @Autowired
+    private MockMvc mockMvc;
+
+    @AfterEach
+    public void resetFirebase() {
+        try {
+            if (deleteUser1) {
+                deleteUser("user1");
+                deleteUser1 = false;
+            }
+        } catch (FirebaseAuthException e) {
+            e.printStackTrace();
+        }
+    }
+
+    @BeforeAll
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Method length is 16 lines (max allowed is 15).](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5213347)